src/field/pokemon.ts

```diff
@@ -1082,8 +1082,12 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
       return true;
     }
 
-    // Final boss does not have passive
-    if (this.scene.currentBattle?.battleSpec === BattleSpec.FINAL_BOSS && this instanceof EnemyPokemon) {
+    // Classic Final boss and Endless Minor/Major bosses do not have passive
+    const waveIndex = this.scene.currentBattle?.waveIndex;
+    if (this instanceof EnemyPokemon &&
+      (this.scene.currentBattle?.battleSpec === BattleSpec.FINAL_BOSS ||
+      this.scene.gameMode.isEndlessMinorBoss(waveIndex) ||
+      this.scene.gameMode.isEndlessMajorBoss(waveIndex))) {
```

Comment:
1. I'm a bit confused why the compiler is not complaining of you throwing a possible `undefined` at `isEndlessMajorBoss` or `isEndlessMinorBoss()` since waveIndex is not guaranteed to be defined (due to the `?` behind `currentBattle`)
2. Little bit readability improvements

```suggestion
    // Classic Final boss and Endless Minor/Major bosses do not have passive
    const { currentBattle, gameMode } = this.scene;
    const waveIndex = currentBattle?.waveIndex;
    if (this instanceof EnemyPokemon &&
      (currentBattle?.battleSpec === BattleSpec.FINAL_BOSS ||
      gameMode.isEndlessMinorBoss(waveIndex) ||
      gameMode.isEndlessMajorBoss(waveIndex))) {
```